pkg/build/controller/build/build_controller.go

```diff
@@ -2333,6 +2333,46 @@ func isValidTransition(from, to buildv1.BuildPhase) bool {
 	return true
 }
 
+func checkLogSnippetsForBadBuildPhase(build *buildv1.Build, update *buildUpdate) bool {
+	switch build.Status.Phase {
+	case buildv1.BuildPhaseFailed:
+		return true
+	case buildv1.BuildPhaseError:
+		// containers completing but exiting with non-0 RCs fall under here
+		return true
+	}
+	if update != nil && update.phase != nil {
+		switch *update.phase {
+		case buildv1.BuildPhaseFailed:
+			return true
+		case buildv1.BuildPhaseError:
+			return true
+		}
+	}
+	return false
+}
+
+func checkLogSnippetForBadPodStatus(pod *corev1.Pod) (bool, bool, int) {
```

Comment:
two bools and an int as return values?  godoc please :)

Reply:
similar comment on "logsnippet" in the name also.

check is more ok as a prefix here since you're returning multiple things, though "get" might be suitable as well.

Reply:
I still went with `isBadPodStatus` in the upcoming push

Reply:
and there is godoc